packages/block-library/src/columns/index.js

```diff
@@ -65,6 +66,27 @@ function getDeprecatedLayoutColumn( originalContent ) {
 	}
 }
 
+/**
+ * Filters the default number of columns of the core/columns block.
+ *
+ * @param {Number} number The column count. Defaults to 2.
+ */
+const DEFAULT_COLUMNS = applyFilters( 'blocks.columns.defaultColumns', 2 );
+
+/**
+ * Filters the minimum number of allowed columns of the core/columns block.
+ *
+ * @param {Number} number The column count. Defaults to 2.
+ */
+const MIN_COLUMNS = applyFilters( 'blocks.columns.minColumns', 2 );
+
+/**
+ * Filters the maximum number of allowed columns of the core/columns block.
+ *
+ * @param {Number} number The column count. Defaults to 6.
+ */
+const MAX_COLUMNS = applyFilters( 'blocks.columns.maxColumns', 6 );
```

Comment:
I just want to mention that in #core-js chats we agreed to limit the use of hooks and filters. We noticed that for the JavaScript code, hooks and filters are not the best extensibility APIs. 

It might be ok for these particular numbers but let's be very careful before adding new ones.

Reply:
That's very good to know, thanks! Do you happen to have a link to that discussion or a summary of it by chance? I'd be curious to learn more.

Reply:
Yes, more context in this summary https://make.wordpress.org/core/2018/11/07/javascript-chat-summary-november-6th/